### ISSUE TITLE
Enabling displays does not work

### ISSUE THREAD

**Issue by novacto3** [falahati/WindowsDisplayAPI#12]

**Comment by falahati** [comment]
Hey,

This is the `master` branch, right? Since the latest release doesn't have the `DisplayScreen` class yet. I will take a look at that. Thanks for reporting.

**Comment by novacto3** [comment]
Yes, sorry for not specifying that. Let me know if I can be of any help with this.

**Comment by novacto3** [comment]
I checked the v1.3.0.13 release and it does not work either. No exception is raised, but the disabled display is not enabled. In v1.2.0.2 it works fine.

**Comment by falahati** [comment]
What you are trying to do? The DriverContext set of APIs is legacy and might not work as expected. Especially if you are trying to enable a device that was connected to a display source or if it was in a clone group. You simply can't do much.

I suggest using classes inside the `DisplayConfig` namespace. Find your physical device using the `PathDisplayTarget` and create a logical desktop corresponding to using the `PathDisplaySource` class then wire them all together and apply the configuration using the `PathInfo` class.

I gave more descriptive information in the following issues:
https://github.com/falahati/NvAPIWrapper/issues/8#issuecomment-465103838
https://github.com/falahati/NvAPIWrapper/issues/26#issuecomment-634543169

**Comment by novacto3** [comment]
I have a display that is "disconnected" (in the terminology of Windows display settings) - it is connected, but disabled. Calling 
`display.Enable(new WindowsDisplayAPI.DisplaySetting(display.GetPreferedSetting()));`
in previous versions of Your library started the display (that is the desired state for me). This functionality is not present in the latest release. Is it a bug or a feature? :-)

**Comment by falahati** [comment]
I don't think I change anything related to this with the last release, give me a moment to check it on my device. So the v1.2 or the v1.1 works but v1.3 fails? What are the values of the unattached display's properties? Does it have a valid display name? ("//DISPLAY1", "//DISPLAY2" ...) And is there an already attached display that has the same name?

**Comment by novacto3** [comment]
I found the bug. In previous versions the attached displays had names "//DISPLAY1", "//DISPLAY2", "//DISPLAY3" and the unattached display had name "//DISPLAY4". In the new release, the attached display names stay the same, but the unattached display has the name "//DISPLAY2" (so there is a duplicity in the naming).

**Comment by falahati** [comment]
Ok, the only change between the version that might be remotely related is the changes to the `GetDisplayDevices()`. I will take another look into that. Meanwhile, if you don't need the DPI functionality you can use v1.2 safely.

Thanks for reporting this issue

**Comment by nxnxfly** [comment]
[DllImport("user32")]
        public static extern Win32Status DisplayConfigGetDeviceInfo(
            ref DisplayConfigSupportVirtualResolution targetSupportVirtualResolution
        );  
Does it work well?  How convert DisplayConfigSupportVirtualResolution to DISPLAYCONFIG_DEVICE_INFO_HEADER   ?

**Comment by nxnxfly** [comment]
[DllImport("user32")]
public static extern Win32Status **DisplayConfigGetDeviceInfo**(
ref DisplayConfigSupportVirtualResolution targetSupportVirtualResolution
);
Does it work well? 

This is the Windows api:
WINUSERAPI LONG WINAPI **DisplayConfigGetDeviceInfo**( Inout_ DISPLAYCONFIG_DEVICE_INFO_HEADER* requestPacket);
And How convert DisplayConfigSupportVirtualResolution to DISPLAYCONFIG_DEVICE_INFO_HEADER ?

**Comment by falahati** [comment]
Yes, that part is correct. It is not the source of the problem.
You see, `DisplayConfigGetDeviceInfo` fills data based on the header of the passed `struct`. So in this case, `DisplayConfigSupportVirtualResolution` is a `struct` that is used to get virtual resolution information. What you are seeing in the native signature as `DISPLAYCONFIG_DEVICE_INFO_HEADER` is actually defined as `DisplayConfigDeviceInfoHeader` in this library and is included as the header of the `DisplayConfigSupportVirtualResolution` to notify the native method of the type of the data requested.

Check here:
https://github.com/falahati/WindowsDisplayAPI/blob/62e27b37dd1a3aa4e49a36286ae39b5aee7f3671/WindowsDisplayAPI/Native/DisplayConfig/Structures/DisplayConfigDeviceInfoHeader.cs#L11
And here:
https://github.com/falahati/WindowsDisplayAPI/blob/62e27b37dd1a3aa4e49a36286ae39b5aee7f3671/WindowsDisplayAPI/Native/DisplayConfig/Structures/DisplayConfigDeviceInfoHeader.cs

**Comment by madmagic007** [comment]
end of 2024, trying to enable a display on the latest nuget package still does not work
